rust/agama-server/src/profile/web.rs

```diff
@@ -0,0 +1,291 @@
+    Ok(output)
+}
+
+#[utoipa::path(
+    post,
+    path = "/autoyast",
+    context_path = "/api/profile",
+    params(ProfileQuery),
+    responses(
+        (status = 200, description = "AutoYaST profile conversion", body = String, content_type = "application/json"),
+        (status = 400, description = "Some error has occurred")
+    )
+)]
+async fn autoyast(
+    query: Query<ProfileQuery>,
+    profile: String, // xml_or_erb_or_empty
+) -> Result<String, ProfileServiceError> {
+    let request_has_body = !profile.is_empty() && profile != "null";
+    if query.url.is_none() || query.path.is_some() || request_has_body {
+        return Err(anyhow::anyhow!(
+            "Only url= is expected, no path= or request body. Seen: url {}, path {}, body {}",
+            query.url.is_some(),
+            query.path.is_some(),
+            request_has_body
+        )
+        .into());
+    }
+
+    let url = Url::parse(query.url.as_ref().unwrap()).map_err(anyhow::Error::new)?;
+    let importer_res = AutoyastProfileImporter::read(&url);
+    match importer_res {
+        Ok(importer) => Ok(importer.content),
+        Err(error) => {
+            // anyhow can be only displayed, not so nice
+            if format!("{}", error).contains("Failed to run") {
+                return Err(make_internal(error));
+            }
+            Err(error.into())
+        }
+    }
+}
+
+#[utoipa::path(
+    post,
+    path = "/execute_script",
+    context_path = "/api/profile",
+    params(ProfileQuery),
+    responses(
+        (status = 200, description = "Script has started"),
+        (status = 400, description = "Some error has occurred")
+    )
+)]
+async fn execute_script(
+    query: Query<ProfileQuery>,
+    script: String, // script_or_empty
+) -> Result<Json<()>, ProfileServiceError> {
+    let request_has_body = !script.is_empty() && script != "null";
+    query.check(request_has_body)?;
+    let script_string = match query.retrieve_profile()? {
+        Some(retrieved) => retrieved,
+        None => script,
+    };
+
+    // write to a temporary path and spawn in background and return
+    // May be long running
+    // TODO: can agama+axum wait for completion and return output?
+    let mut named_tempfile = tempfile::Builder::new()
+        .prefix("agama-script")
+        .permissions(std::fs::Permissions::from_mode(0o700))
+        .tempfile()
+        .context("Creating temporary file for script")?;
+    named_tempfile
+        .as_file_mut()
+        .write_all(script_string.as_bytes())
+        .context("Writing script text")?;
+    // close the file otherwise exec fails with ETXTBSY
+    let path = named_tempfile.into_temp_path();
+    let path = path.keep().context("Persisting script file")?;
+
+    let mut child = std::process::Command::new(&path)
+        .spawn()
+        .context(format!("Spawning script {:?}", path))?;
+
+    // Do not child.wait() for the script to finish,
```

Comment:
As discussed in IRC, I am not sure about this behavior. I would expect to have always the same answer, no matter whether the call takes 50ms or 60ms. I would go for:

1) Waiting until it finishes but without blocking the server. That's why I mentioned using `tokio::process::Command` instead (but I have not tried if it works as expected). We could even move the script to a separate Tokio task.
2) Always return a 200 and move the script executation to a separate Tokio task.

We even discussed about having an API to run jobs. You would create a job by calling `POST /jobs`, which would return a job ID. So you can check the current status with `GET /jobs/1`. Of course, we could integrate this API with the events system too. But I would say it is out of scope.